modules/mill-codegen-plugin/src/smithy4s/codegen/mill/Smithy4sExternalModule.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.codegen.mill
+
+import mill.eval.Evaluator
+import mill.define.ExternalModule
+import mill.define.Target
+import coursier.maven.MavenRepository
+import smithy4s.codegen.SmithyBuildJson
+import smithy4s.codegen.SmithyBuildData
+import mill.define.Command
+import mill.api.PathRef
+
+object Smithy4sExternalModule extends ExternalModule {
```

Comment:
when creating mill external module, it's important to think about the UX imposed onto the users, as they don't get autocompletion. You want the command to be as easy to remember as possible. I think it should be something like : 

`smithy4s.LSP/config`

Reply:
That would mean moving the object under the `smithy4s` package, right?

Reply:
would `smithy4s.codegen.LSP/smithyBuild` be fine with you?

I feel like we're gonna add other commands in the future in this module and `config` sounds too generic

I added `smithy4s.codegen` so package visiblity of the `SmithyBuildJson` still make some sense

Reply:
package is fine. For reference, similar commands in the mill ecosystem are : 

``` 
mill.idea.GenIdea/idea 
mill.contrib.bloop.Bloop/install
```

I think the first one is silly, there's 3 repetitions of `idea`. I however like the `GenIdea` object
I like `install` but it's a bit generic. 

Combining the two, we could have something like : 

``` 
smithy4s.codegen.LSPConfig/update
smithy4s.codegen.LSP/updateConfig
```

Reply:
done, I've added some docs in the editor support page as well